test/test_automl/test_estimators.py

```diff
@@ -671,6 +673,33 @@ def test_conversion_of_list_to_np(self, fit_ensemble, refit, fit):
         self.assertEqual(fit_ensemble.call_count, 1)
         self.assertIsInstance(fit_ensemble.call_args[0][0], np.ndarray)
 
+    def test_classification_pandas_support(self):
+        X, y = sklearn.datasets.fetch_openml(
+            data_id=37,  # diabetes
+            return_X_y=True,
+            as_frame=True,
+        )
+        # This test only make sense if input is dataframe
+        self.assertTrue(isinstance(X, pd.DataFrame))
+        self.assertTrue(isinstance(y, pd.Series))
+        automl = AutoSklearnClassifier(
+            time_left_for_this_task=30,
+            per_run_time_limit=5,
+        )
+
+        automl.fit(X, y)
+
+        # Make sure that at least better than random.
+        # We use same X_train==X_test to test code quality
+        self.assertTrue(automl.score(X, y) > 0.5)
```

Comment:
Hey, I just checked the dataset and diabetes appears to be suboptimal in this setting as it contains only numerical attributes. Could you please try dataset 2 as this contains almost everything that makes Auto-sklearn's life hard (lots of missing values, categorical features, boolean features, features which have only one value or are missing). Also, both datasets are unbalanced, so a score larger than 0.5 won't be sufficient here. Could you please try with for example AUC as the metric will be more useful there?

Reply:
Moved to this dataset! Nevertheless, I have to remove the columns that are only Nan!